File: solaris-ecommerce-api/src/main/java/com/luccavergara/solaris/ecommerce/entity/Customer.java
package com.luccavergara.solaris.ecommerce.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "customers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String razonSocial;

    @Column(nullable = false, unique = true, length = 20)
    private String cuit;

    private String email;

    private String phone;

    private String address;

    private String city;

    private String province;

    private String postalCode;

    private String country;

    @Column(length = 50)
    private String taxCondition;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @Column(nullable = false)
    @Builder.Default
    private Boolean active = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_user_id")
    private User createdBy;
}

File: solaris-ecommerce-web/src/types/customer.ts
export type DocumentType = 'CUIT' | 'CUIL' | 'DNI'

export type CondicionIva =
  | 'RESPONSABLE_INSCRIPTO'
  | 'MONOTRIBUTO'
  | 'EXENTO'
  | 'CONSUMIDOR_FINAL'
  | 'NO_CATEGORIZADO'
  | 'RESPONSABLE_NO_INSCRIPTO'

export type CustomerDocument = {
  id?: number
  documentType: DocumentType
  documentNumber: string
  primary?: boolean
}

export type Customer = {
  id: number
  razonSocial: string
  cuit: string
  email: string | null
  phone: string | null
  address: string | null
  city: string | null
  province: string | null
  postalCode: string | null
  country: string | null
  taxCondition: CondicionIva
  active?: boolean
  userId?: number
  createdAt: string
}

export type CustomerRequest = {
  razonSocial: string
  cuit: string
  email?: string
  phone?: string
  address?: string
  city?: string
  province?: string
  postalCode?: string
  country?: string
  taxCondition: CondicionIva
}
